src/azure-cli/azure/cli/command_modules/network/custom.py

```diff
@@ -6657,6 +6664,11 @@ def list_traffic_manager_endpoints(cmd, resource_group_name, profile_name, endpo
     return [e for e in profile.endpoints if not endpoint_type or e.type.endswith(endpoint_type)]
 
 
+def check_traffic_manager_name(cmd, client, name, resource_type):
```

Comment:
If you want to make `resoure_type` unrequired, you can change like following
```suggestion
def check_traffic_manager_name(cmd, client, name, resource_type=None):
```

Reply:
> This commit is not correct this makes resource_type parameter as required. This will cause the breaking change. Keeping ignore_type is correct way where this parameter is set as non required.
> 
> ![image](https://user-images.githubusercontent.com/63234605/150982076-8714a49f-68cb-4740-ad39-6d49cd9faa2e.png)

I'm confusing, can user pass --type argument for this command?

Reply:
resource_type is set to default='Microsoft.Network/trafficManagerProfiles'.
We don't need user to input this field. Its not required field. and we do not expect user to enter it as well.

If we set it to None then it will be incorrect.

Reply:
We renamed the variable type to resource_type because of style issue. because of which build was failing.

Reply:
![image](https://user-images.githubusercontent.com/63234605/150986323-fa98e90c-4208-470b-af7f-052a588cfc49.png)

This code already take care of it making as non required.